src/redirects-manager.tsx

```diff
@@ -1,238 +1,60 @@
-
-        this.createRedirection = () => {
-            let newState = this.state;
-            newState.store.push({id: v4()});
-            this.setState(newState);
-        }
-
-        this.toggleWildcard = (e) => {
-            let newState = this.state;
-            newState.wildcard = !this.state.wildcard;
-            this.setState(newState);
-        }
+interface RedirectsManagerProps  {
+    initialState: RedirectsManagerStateInterface;
+}
 
-        this.saveStore = async () => {
-            this.setState({saving: true});
-            const payload = {wildcard: this.state.wildcard, store: this.state.store}
-            const init = {
-                method: 'POST',
-                headers: {
-                    'Content-Type': 'application/json'
-                },
-                body: JSON.stringify(payload)
-            }
-            let result: Response;
-            try {
-                result = await fetch(ajaxUrl+'?action=saveRedirects', init);
-            } catch (e) {
-                this.showNotification('error', 'An Error ocurred! Changes not saved');
-                return e;
-            }
-            if (result.ok) {
-                let json: {data : {redirects_added: number, redirects_modified: number, redirects_deleted: number, store: Array<WerRedirectionData>}};
-                try {
-                    json = await result.json();
-                } catch (e) {
-                    this.showNotification('error', 'An Error ocurred! Changes not saved');
-                    return e;
-                }
-                if (json.data.redirects_added === 0 && json.data.redirects_modified === 0 && json.data.redirects_deleted === 0)
-                {
-                    this.showNotification('warning', 'No changes were made!');
-                } else {
-                    this.showNotification('success', ({msg}) => { 
-                        return (
-                        <div>Redirects Succesfully saved!<br/>
-                        Added: {json.data.redirects_added}<br/>
-                        Modified: {json.data.redirects_modified}<br/>
-                        Deleted: {json.data.redirects_deleted}</div>
-                        )
-                    });
-                }
-                let newState = this.state;
-                newState.store = this.validateStore(json.data.store);
-                newState.saving = false;
-                newState.lastSave = result.ok;
-                this.setState(newState);
-            } else {
-                this.showNotification('error', 'An Error ocurred! Changes not saved');
-            }
-            return;
-        }
+export const RedirectsManager = ({initialState}: RedirectsManagerProps) => {
+    const [state, dispatch] = React.useReducer(redirectsManagerReducer, initialState);
+    return (
+        <div>
+            <table className="widefat" style={{width: '100%'}}>
+                <Toolbar dispatch={dispatch} state={state} />
```

Comment:
i think we agreed on passing all items through the toolbar / filter first and use the filtered props to feed the List and the Pagination.
Also the `<table>` should not wrap the toolbar, but be contained instead.

Reply:
As I said on a previous comment, the reason is that Toolbar was not for filtering only, but in any case, I think is doable if you want it

Reply:
applied